File: DekafTestDB/src/core/H2dbTestHelper.java
package org.jetbrains.dekaf.core;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.dekaf.text.Scriptum;



/**
 * @author Leonid Bushuev from JetBrains
 **/
public class H2dbTestHelper extends BaseTestHelper<DBFacade> {

  public H2dbTestHelper(@NotNull final DBFacade db) {
    super(db, Scriptum.of(H2dbTestHelper.class));
  }

  @Override
  public void prepareX1() {
    performCommand("create or replace view X1 as select 1 as X from dual");
  }

  @Override
  public void prepareX1000() {
    performCommand(scriptum, "X1000");
  }

  @Override
  public void prepareX1000000() {
    prepareX1000();
    performCommand(scriptum, "X1000000");
  }
}

File: dekaf-test-db/src/core/H2dbTestHelper.kt
package org.jetbrains.dekaf.core


/**
 * H2db test helper.
 *
 * @author Leonid Bushuev
 */
class H2dbTestHelper(db: DBFacade) : BaseTestHelper(db, H2dbTestHelper::class) {

    override fun prepareX1() {
        performCommand("create or replace view X1 as select 1 as X from dual")
    }

    override fun prepareX1000() {
        performCommand(scriptum, "X1000")
    }

    override fun prepareX1000000() {
        prepareX1000()
        performCommand(scriptum, "X1000000")
    }

}
